reconcile/utils/oc_connection_parameters.py

```diff
@@ -105,13 +125,19 @@ def from_cluster(
         if cluster_admin:
             if cluster.cluster_admin_automation_token:
                 try:
-                    cluster_admin_automation_token = secret_reader.read_secret(
+                    secret_raw = secret_reader.read_all_secret(
                         cluster.cluster_admin_automation_token
                     )
                 except SecretNotFound:
                     logging.error(
                         f"[{cluster.name}] admin token {cluster.cluster_admin_automation_token} not found"
                     )
+                else:
+                    cluster_admin_automation_token = (
+                        OCConnectionParameters._get_token_verify_server_url(
+                            ClusterSecret(**secret_raw), cluster
+                        )
+                    )
```

Comment:
nit: wondering if we require a separate else block here. My understanding is it cannot raise a `SecretNotFound` error, so it cannot accidentally be caught. Not 100% sure, but I think in this case it is safe to put this into the try block too?

Reply:
Just trucked along with the existing pattern. Found it confusing as well. Can do a refactoring here